### ISSUE TITLE
Yolov8: to be or not to be augmented?

### ISSUE THREAD

**Issue by lesept777** [ultralytics/ultralytics#8794]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I'd like to train my network without any data augmentation.
When I do:
`results = model.train(data='./data', epochs=100, imgsz=32*22, cos_lr=True, lr0=0.001, batch=-1, optimizer='AdamW')`
I can see a comment saying 'augment=false' (or similar). But when I open the train_batch0.jpg image, I can see that there were some transformations applied.
![train_batch0](https://github.com/ultralytics/ultralytics/assets/30208898/c8e54778-9595-4653-a391-e4cbd606143a)
So what is true: transformations or not?

How can I specify that I don't want any transformation or augmentation in the model.train command?
Thansk for your help

**Comment by glenn-jocher** [comment]
@lesept777 hello! 😊

It looks like you're aiming to train your model without any data augmentation. In YOLOv8, certain augmentations are applied by default to improve model robustness. However, if you wish to disable these augmentations, you can do so by setting the `augment` argument to `False` in your `model.train()` command.

Here's how you can specify no transformations or augmentations:

```python
results = model.train(data='./data', epochs=100, imgsz=32*22, cos_lr=True, lr0=0.001, batch=-1, optimizer='AdamW', augment=False)
```

Just adding `augment=False` should do the trick! Keep in mind, though, that training without any data augmentation might affect the model's ability to generalize well to unseen data.

If you have any more questions or need further assistance, feel free to ask. Happy training! 🚀

**Comment by lesept777** [comment]
Thanks a lot Glenn, as my images are spectrograms, augmentation as seen in the picture above does not always make sense. Are there any augmentations that would be interesting in this case?

**Comment by glenn-jocher** [comment]
@lesept777 hey there! 😊

For spectrogram images, you're right that not all standard augmentations would be beneficial. However, considering the nature of spectrograms, augmentations that do not distort the frequency-time relationship might still be useful. For instance, you might consider:

- **Random Cropping** (if it doesn't remove important parts of your data)
- **Horizontal Flipping** (if your data's interpretation isn't direction-dependent)
- **Brightness & Contrast Adjustments** (to simulate variations in recording conditions)

Regarding the `augment=False` setting not seeming to take effect, it's possible that some default settings or behaviors are still applying augmentations. The `auto_augment=randaugment` and other augmentation parameters in your output suggest that augmentations are still being applied despite the `augment=False` flag.

To ensure no augmentation is applied, you might want to double-check your dataset configuration file (`data_div3.yaml`) to make sure there are no augmentation settings enabled there. Also, ensure that the `augment=False` parameter is correctly passed where it's supposed to be.

If you're still seeing unexpected behavior, it might be worth raising this as a potential issue on the GitHub repository for further assistance. The YOLOv8 team is always keen to help out! 🚀

**Comment by glenn-jocher** [comment]
@lesept777 hello! 😊

The dataset configuration file (`data_div3.yaml`) you mentioned should be created by you as part of your dataset preparation. It's a YAML file that specifies paths to your training and validation data, class names, and potentially other settings. Since you've created the dataset yourself, you'll also need to create this file. Here's a simple example:

```yaml
# data_div3.yaml
train: ./data_div3/train/images  # path to training images
val: ./data_div3/val/images  # path to validation images
nc: 10  # number of classes
names: ['class0', 'class1', ..., 'class9']  # class names
```

Regarding the `augment=False` parameter, you've placed it correctly in your `model.train()` command. This should ideally disable augmentations during training. If you're still seeing unexpected augmentations, it might be due to default behaviors or settings elsewhere in the training process. 

If you've confirmed that your `data_div3.yaml` doesn't specify any augmentations and you're still encountering issues, it might be worth checking the documentation or reaching out for further support. Keep up the great work! 🚀

**Comment by lesept777** [comment]
Hi Glenn
I have created a yaml file as you suggested. I placed it in the same folder as my python code.
```
# data_div3.yaml

# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: ./data_div3/
train: train/images  # path to training images (relative to 'path') 172757 images
val:   val/images    # path to validation images                     34550 images
test:  val/images    # path to validation images
nc: 10  # number of classes
#names: ['airport', 'bus', 'metro', 'metro_station', 'park', 'public_square', 'shopping_mall', 'street_pedestrian', 'traffic', 'tram']  # class names
names:
    0: airport
    1: bus
    2: metro
    3: metro_station
    4: park
    5: public_square
    6: shopping_mall
    7: street_pedestrian
    8: traffic
    9: tram
```
I still see the same behavior:
```
Ultralytics YOLOv8.1.19 🚀 Python-3.8.10 torch-2.0.1+cu117 CUDA:0 (NVIDIA A100-PCIE-40GB, 40536MiB)
engine/trainer: task=classify, mode=train, model=yolov8n-medium.yaml, data=./data_div3, epochs=50, time=None, patience=100, batch=-1, imgsz=300, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train6, exist_ok=False, pretrained=True, optimizer=AdamW, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=True, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, **augment=False**, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.002, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, **auto_augment=randaugment**, erasing=0.4, crop_fraction=1.0, cfg=None, tracker=botsort.yaml, save_dir=runs/classify/train6
```
and train_batch0.jpg image still similar to the one I put above.
What are the augmentation related parameters in the above parameters list?

**Comment by glenn-jocher** [comment]
@lesept777 hey there! 👋

Great job on setting up your `data_div3.yaml` file! It looks well-organized. Regarding the augmentations, it seems like the `augment=False` parameter is correctly set to disable augmentations during training. However, the presence of `auto_augment=randaugment` in your output suggests that some form of augmentation is still being applied.

To ensure no augmentations are applied, you might want to explicitly set `auto_augment=None` in your training command, like so:

```python
results = model.train(data='./data_div3', epochs=50, imgsz=300, cos_lr=True, lr0=0.001, batch=-1, optimizer='AdamW', augment=False, auto_augment=None)
```

This should help override any default augmentation settings. If you're still encountering unexpected behavior, it might be worth revisiting the documentation or reaching out for further support. Keep up the great work! 🚀

**Comment by lesept777** [comment]
Thanks Glenn
Here is what I get with your parameters:

> YOLOv8n-medium summary: 81 layers, 398554 parameters, 398554 gradients, 2.5 GFLOPs
> Ultralytics YOLOv8.1.19 🚀 Python-3.8.10 torch-2.0.1+cu117 CUDA:0 (NVIDIA A100-PCIE-40GB, 40536MiB)
> engine/trainer: task=classify, mode=train, model=yolov8n-medium.yaml, data=./data_div3, epochs=50, time=None, patience=100, batch=-1, imgsz=320, save=True, save_period=-1, cache=False, device=None, workers=8, project=None, name=train10, exist_ok=False, pretrained=True, optimizer=AdamW, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=True, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=False, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, **augment=False**, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, show_boxes=True, line_width=None, format=torchscript, keras=False, optimize=False, int8=False, dynamic=False, simplify=False, opset=None, workspace=4, nms=False, lr0=0.002, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=7.5, cls=0.5, dfl=1.5, pose=12.0, kobj=1.0, label_smoothing=0.0, nbs=64, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, **translate=0.1, scale=0.5**, shear=0.0, perspective=0.0, flipud=0.0, **fliplr=0.5**, mosaic=1.0, mixup=0.0, copy_paste=0.0, **auto_augment=None**, **erasing=0.4**, **crop_fraction=1.0**, cfg=None, tracker=botsort.yaml, save_dir=runs/classify/train10
> 

I believe the parameters in bold may be related to data augmentation. How can I set them? Should I do this in the yaml file?
Now, the train_batch0.jpg image looks like this:

![train_batch0](https://github.com/ultralytics/ultralytics/assets/30208898/1a7426d4-9ed0-431c-bc39-73f896d29a55)

So there is still one option to set somewhere to remove these black windows. Is it `crop_fraction`, or maybe `erasing`?

**Comment by lesept777** [comment]
Thanks
I guess that setting
`translate=0, scale=0, fliplr=0, erasing=0, crop_fraction=0`
would remove all those augmentations.

**Comment by glenn-jocher** [comment]
Absolutely, @lesept777! 😊 Your settings look spot on for disabling those augmentations:

```python
translate=0, scale=1, fliplr=0, erasing=0, crop_fraction=1
```

Just a tiny adjustment: setting `scale` to `1` (instead of `0`) keeps the original image scale unchanged. This should effectively turn off the augmentations you're concerned about. Let us know if you need further assistance!

**Comment by arasul42** [comment]
1. What are the args are affected by setting augment: false. 
2. If I want to set all augmentation to None and fine tune only one (for example scale) what is the way forward. 
Tnanks in advance for the help.

**Comment by glenn-jocher** [comment]
Setting `augment=False` in YOLO training primarily affects the default augmentation settings, such as translation, scaling, and flipping, effectively disabling them. To disable all augmentations and fine-tune only one, like `scale`, you can manually set other augmentation parameters to their neutral values (e.g., `translate=0`, `fliplr=0`, etc.) and adjust `scale` as needed in your training command.

**Comment by arasul42** [comment]
Thanks Glenn, much helpful. If auto_augment is set to 'randomaugment' and I set scale to some value (i.e 0,5) does it mean auto augmentation will be applied upto 0.5 scale value? Advance thanks for your reply.

**Comment by glenn-jocher** [comment]
Auto augmentation policies like `randaugment` operate independently of individual augmentation parameters such as `scale`. If `auto_augment` is set to `randaugment`, it will apply its predefined augmentation strategies, and any specific `scale` setting won't directly influence these strategies. If you want precise control over `scale`, consider setting `auto_augment` to `None` and then adjusting `scale` as needed.

**Comment by arasul42** [comment]
ef plot_initial_pos(run_folder, initial_positions, ideal_x, ideal_y):
    """Plot and save initial positions."""
    initial_x, initial_y, initial_z = zip(*initial_positions)
    size_based_on_z = 3000 / (1 + abs(np.array(initial_z) - 35))

    plt.figure(figsize=(10, 8))
    scatter = plt.scatter(
        initial_x, initial_y,
        c=abs(np.array(initial_z) - 35),
        cmap='Blues_r',
        s=size_based_on_z,
        edgecolor='black',
        linewidth=0.5,
        label='Initial Positions'
    )
    plt.scatter(ideal_x, ideal_y, color='red', s=600, marker='o', edgecolor='black', linewidth=0.5, label='Ideal Landing Position')
    plt.colorbar(scatter, label='Height Difference from Z=35 (m)', orientation='vertical')
    plt.title("Initial Positions with Dynamic Size and Color Scaling")
    plt.xlabel("X Position (m)")
    plt.ylabel("Y Position (m)")
    plt.legend(loc='upper center', bbox_to_anchor=(0.5, -0.1), ncol=2, borderaxespad=0.)
    plt.grid(True)
    plt.xlim(ideal_x - 50, ideal_x + 50)
    plt.ylim(ideal_y - 50, ideal_y + 50)
    plt.tight_layout()
    plt.savefig(f"{run_folder}/initial_position/initial_positions.png")
    plt.close()

def plot_final_pos(run_folder, final_positions, ideal_x, ideal_y):
    """Plot and save final positions."""
    final_x, final_y, final_z = zip(*final_positions)

    plt.figure(figsize=(10, 8))
    scatter_final = plt.scatter(
        final_x, final_y,
        c=abs(np.array(final_z) - 35),
        cmap='Greens',
        s=100,
        edgecolor='black',
        linewidth=0.5,
        label='Final Positions'
    )
    plt.scatter(ideal_x, ideal_y, color='red', s=600, marker='o', edgecolor='black', linewidth=0.5, label='Ideal Landing Position')
    plt.colorbar(scatter_final, label='Height Difference from Z=35 (m)', orientation='vertical')
    plt.title("Final Positions with Color Scaling Only (Fixed Size)")
    plt.xlabel("X Position (m)")
    plt.ylabel("Y Position (m)")
    plt.legend(loc='upper center', bbox_to_anchor=(0.5, -0.1), ncol=2, borderaxespad=0.)
    plt.grid(True)
    plt.xlim(ideal_x - 50, ideal_x + 50)
    plt.ylim(ideal_y - 50, ideal_y + 50)
    plt.tight_layout()
    plt.savefig(f"{run_folder}/landing_position/final_positions.png")
    plt.close()


give me a script that takes a csv as input and give the above plots. CSV structure is below: 

x_i,y_i,z_i,Time to Land (s),Landing Result (Success/Fail),x_final,y_final,z_final,final_roll,final_pitch,final_yaw
-86.9701342713033,44.26729079419679,81.34827730917645,90.06327414512634,Success,-79.51906275746946,73.99910000786251,33.20424730709552,,,
-85.30321688841347,96.44747096268013,147.90570775200928,21.384795427322388,Fail,-85.297501609755,96.35539776855609,147.24372839136254,,,
-55.03550157517163,95.59808105272717,60.63939042589922,98.40596055984497,Fail,-81.64417695111061,70.70900696474713,32.78495551324631,,,
-100.2759453770804,58.321820659957886,79.81337505894638,81.54867696762085,Fail,-82.03926270260555,72.86502814973194,34.44944021502708,,,
-104.54547730130602,88.52626372454375,59.76669489934788,46.958561420440674,Success,-78.08406082776672,75.72723791637148,32.671668816421565,3.1363139484008045,-0.024855457695011404,-0.06105608558186671
-101.1874285049216,59.270003825544265,74.06080391020498,79.71562433242798,Success,-79.87952973469513,72.98948529428048,33.64937297785017,,,
-76.58661387766378,89.5654096959326,64.01531482518108,71.43733787536621,Success,-80.35776099047794,73.24872551812241,33.44255643002601,,,
-51.11565178107013,51.950573772872055,78.89568089991424,103.52459239959717,Fail,-68.81770511010588,69.65570130894096,53.08849740057414,,,
-106.33181667785978,65.28571961425678,63.585847671652814,46.77566885948181,Fail,-98.45760462409575,72.82548207527387,55.5574496352687,,,
-75.9310117375246,89.55844255535968,56.628685132380376,59.44422912597656,Success,-82.94147898076916,73.58629581153725,33.0484090579144,,,
